File: 0463_island_perimeter/island_perimeter.go
package main

func islandPerimeter(grid [][]int) int {
	r, c, res := len(grid), len(grid[0]), 0
	for i := 0; i < r; i++ {
		for j := 0; j < c; j++ {
			if grid[i][j] == 1 {
				res += 4
				if i > 0 && grid[i-1][j] == 1 {
					res -= 2
				}
				if j > 0 && grid[i][j-1] == 1 {
					res -= 2
				}
			}
		}
	}
	return res
}

func main() {

}

File: 0463_island_perimeter/island_perimeter.py
from typing import List


def island_perimeter(grid: List[List[int]]) -> int:
    """
    Time: O(m * n)
    Space: O(1)
    """
    r, c, res = len(grid), len(grid[0]), 0
    for i in range(r):
        for j in range(c):
            if grid[i][j]:
                res += 4
                if i and grid[i - 1][j]:
                    res -= 2
                if j and grid[i][j - 1]:
                    res -= 2
    return res
